index.bs

```diff
@@ -489,18 +508,17 @@ An {{MLContext}} interface represents a global state of neural network execution
 
 In a situation when a GPU context executes a graph with a constant or an input in the system memory as an {{ArrayBufferView}}, the input content is automatically uploaded from the system memory to the GPU memory, and downloaded back to the system memory of an {{ArrayBufferView}} output buffer at the end of the graph execution. This data upload and download cycles will only occur whenever the execution device requires the data to be copied out of and back into the system memory, such as in the case of the GPU. It doesn't occur when the device is a CPU device. Additionally, the result of the graph execution is in a known layout format. While the execution may be optimized for a native memory access pattern in an intermediate result within the graph, the output of the last operation of the graph must convert the content back to a known layout format at the end of the graph in order to maintain the expected behavior from the caller's perspective.
 
-When an {{MLContext}} is created with {{MLContextOptions}}, the user agent selects and creates the underlying execution device by taking into account the application's [=power preference=] and [=device preference=] specified in the {{MLPowerPreference}} and {{MLDevicePreference}} options.
+When an {{MLContext}} is created with {{MLContextOptions}}, the user agent selects and creates the underlying execution device by taking into account the application's [=power preference=] and [=device type=] specified in the {{MLPowerPreference}} and {{MLDeviceType}} options.
 
-The following table summarizes the types of resource supported by the device selected.
+The following table summarizes the types of resource supported by the device option.
 
 <div class="note">
 <table>
-  <tr><th>Device Type<th>ArrayBufferView<th>GPUBuffer<th>GPUTexture<th>WebGLBuffer<th>WebGLTexture
-  <tr><td>GPUDevice<td>Yes<td>Yes<td>Yes<td>No<td>No
-  <tr><td>WebGLRenderingContext<td>Yes<td>No<td>No<td>Yes<td>Yes
-  <tr><td>default<td>Yes<td>No<td>No<td>No<td>No
-  <tr><td>gpu<td>Yes<td>No<td>No<td>No<td>No
+  <tr><th>Device Option<th>ArrayBufferView<th>GPUBuffer<th>GPUTexture<th>WebGLBuffer<th>WebGLTexture
   <tr><td>cpu<td>Yes<td>No<td>No<td>No<td>No
+  <tr><td>gpu (GPUDevice == null)<td>Yes<td>No<td>No<td>No<td>No
+  <tr><td>gpu (GPUDevice != null)<td>Yes<td>Yes<td>Yes<td>No<td>No
```

Comment:
If the `MLContext` is created from a `GPUDevice`, I would suggest it only accepts the `GPUBuffer` and `GPUTexture`. This design would help simplify the spec / implementation of WebNN API and leave the `ArrayBufferView` uploading / readback to WebGPU API where it is already well defined.